test/common/http/conn_pool_grid_test.cc

```diff
@@ -227,6 +231,38 @@ TEST_F(ConnectivityGridTest, TimeoutThenSuccessParallelFirstConnects) {
   EXPECT_NE(grid_.callbacks(0), nullptr);
   EXPECT_CALL(callbacks_.pool_ready_, ready());
   grid_.callbacks(0)->onPoolReady(encoder_, host_, info_, absl::nullopt);
+  EXPECT_FALSE(grid_.isHttp3Broken());
+}
+
+// Test both connections happening in parallel and the second connecting before
+// the first eventually fails..
+TEST_F(ConnectivityGridTest, TimeoutThenSuccessParallelSecondConnectsFirstFail) {
```

Comment:
should we test the other order as well, of if it's already tested, just verify H3 gets set to broken?

Reply:
I *think* this is what happens in the existing TimeoutThenSuccessParallelSecondConnects test. I think that starts H3, times out, starts TCP, H3 fails (but doesn't up call), TCP succeeds (and upcalls) and we check that H3 is broken.

Does that sound right? Or was there a different ordering you were thinking about?